boot/mbed/mcuboot_main.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2020 Embedded Planet
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+#if MCUBOOT_BOOTLOADER_BUILD
+
+#include <stdlib.h>
+#include "bootutil/bootutil.h"
+#include "bootutil/image.h"
+#include "hal/serial_api.h"
+#include "mbedtls/platform.h"
+#include "mbed_application.h"
+
+#define MBED_TRACE_MAX_LEVEL TRACE_LEVEL_DEBUG
+#define TRACE_GROUP "BL"
+#include "mbed-trace/mbed_trace.h"
+
+int main()
+{
+    mbed_trace_init();
+
+    tr_info("Starting MCUboot");
+
+    // Initialize mbedtls crypto for use by MCUboot
+    mbedtls_platform_context unused_ctx;
+    int rc = mbedtls_platform_setup(&unused_ctx);
+    if(rc != 0) {
+        tr_error("Failed to setup Mbed TLS, error: %d", rc);
+        exit(rc);
+    }
+
+    struct boot_rsp rsp;
+    rc = boot_go(&rsp);
+    if(rc != 0) {
+        tr_error("Failed to locate firmware image, error: %d", rc);
+        exit(rc);
+    }
+
+    uint32_t address = rsp.br_image_off + rsp.br_hdr->ih_hdr_size;
+
+    tr_info("Booting firmware image at 0x%x", address);
```

Comment:
There is no longer a need to add `\n`?

Reply:
No need for stdlib-based printing - mbed-trace uses `puts()` from stdlib (like `printf()`) instead of raw serial. Some flushing mechanism is in place I guess.

Reply:
Even though it is printed, it doesn't add a newline... need to add the workaround back.

Reply:
Added back